### DISCUSSION TITLE
Using Polarfire "MPFS250-Video-Kit", we want 4k/60fps video streaming with FPGA H.264 Video Streaming, but we currently have only 720p video output.

### DISCUSSION THREAD

**Discussion by hiteshmathu** [polarfire-soc/polarfire-soc-documentation#413] [Applications and demos]
Our Goal: Using "MPFS250-Video-Kit", we want 4k/60fps video streaming with FPGA H.264 Video Streaming, but we currently have only 720p video output.

Issues:

1) To achieve the above output, I updated the GitHub code [MPFS_VIDEO_KIT_H264_REFERENCE_DESIGN.tcl] i.e changed the frequency of IMX from 170 Mhz to 148.5 Mhz and image processing IP using "Libero soc" and flash .job file on the kit but I did not receive any data. so, what changes are needed to get 4k/60fps with H264?

Code that I used to get the above output:

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/7fe86ab7-ced4-4b01-aac0-b4634dc0ead0)

Updated Code changes:
1) IMX334_IF_TOP [IP]

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/6aea8158-3521-44fd-95ae-a62a424b2924)

Error Message:

 from element /GstPipeline:pipeline0/GstV4l2Src:v4l2src0: Cannot identify device '/dev/video0'.

2) To resolve the above error when I flashed the new Linux image i.e. given on the application note [Download the Linux .wic image.] on the kit, I received the below error message. which Linux prebuild image is required to get the above output?

An image that I flashed on the kit:

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/52f3d4d4-7218-4100-a13c-151ce141d3f7)

Error Message:

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/5eded049-3ce2-40f8-8007-bf97c11d7ae6)

Need help on this... i m stuck with it.. TIA

**Comment by sc-mchp** [comment]
Hi Hiteshmathu,

If your using latest TCL from here  https://github.com/polarfire-soc/polarfire-soc-video-kit-reference-design
you should use the later version yocto image "https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/tag/v2024.02"

Thanks,
Shravan

**Comment by hiteshmathu** [comment]
Hi shravanI35088, 

can you suggest a clock structure/changes/prebuilt image/Documentation for FPGA that supports 4k/60fps with H264 streaming?

**Reply by sc-mchp** [reply]
1) H.264 4k/60FPS not supported as of now

2) for clock changes refer MIPI CSI2 Rx user guide
https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/UserGuides/microchip_mipi_csi_2_receiver_decoder_ip_user_guide_ug0806_v8.pdf

3) yocto wic (https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/download/v2024.02/core-image-minimal-dev-mpfs-video-kit-20240221164152.rootfs.wic.gz) 

4) jobfile (https://github.com/polarfire-soc/polarfire-soc-video-kit-reference-design/releases/download/v2024.02/MPFS_VIDEO_KIT_BASE_DESIGN_2024_02.zip)

5) User guides:
https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/mpfs-video-kit-user-guide.md

https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/updating-mpfs-kit.md

**Reply by hiteshmathu** [reply]
when i flash above linux image i m getting below error,

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/e2b54c87-1f57-49d9-99b0-11bfe4098b43)

**Reply by sc-mchp** [reply]
please use prebuild job file from the link given.

**Reply by hiteshmathu** [reply]
I flashed the above linux image and .job file but gst command not supported....
gst-launch-1.0: command not found ...............why?
![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/15924135/7d96dd44-ed58-4962-b61d-d2c9f90f867d)

**Reply by sc-mchp** [reply]
gst-launch-1.0 is not supported as of now. if you want you can build with yocto build system.

**Reply by hiteshmathu** [reply]
What are the other way if gst-launch-1.0 is not supported? 
in the default Linux image[i.e already come with kit], gst-launch-1.0 is supported but why in latest image it does not supported?

**Reply by sc-mchp** [reply]
we will add soon in the release images.
we have not used gst-launch-1.0 in the demo so removes.

**Comment by hiteshmathu** [comment]
Hi shravanI35088,

Can you provide the image that supports gst-launch-1.0?
Or what are the other ways using which I can stream the video?

**Comment by eppidei** [comment]
Hi @hiteshmathu , I guess you would like to control the video pipeline using gstreamer. Is that correct? That is not supported at the moment.

**Comment by hiteshmathu** [comment]
Hi all,
How can I stream 4k/60 fps using Polarfire  Polarfire "MPFS250-Video-Kit"?

**Reply by sc-mchp** [reply]
4k/60fps not supported as of now ...

**Comment by hiteshmathu** [comment]
Hi, 

Which third-party "H.264 4k Decoder" support for this kit?
